### PR TITLE
chore: proper new arch + backwards compat using install function

### PR DESCRIPTION
**Pull request by hannojg**
See: https://reactnative.dev/blog/2024/08/12/release-0.75#accessing-jsiruntime-in-turbomodules

- Updates the example app to be RN 0.75
- Removes being a Cxx Only turbo module / enables this to be a prefab again on android
- Works on old arch / new arch

---

All tests passing on new and old arch

Screenshot_20241016_160635
Screenshot 2024-10-16 at 16 24 16

### CODE REVIEW

File: package/ios/NativeWorklets.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+//  NativeWorklets.h
+//  Pods
+//
+//  Created by Hanno Gödecke on 16.10.24.
```
Comment (mrousavy) [inline package/ios/NativeWorklets.h:5]: credits? 🥺

**Comment by mrousavy** [comment]
looks good 👍

File: package/ios/NativeWorklets.h (hunk 1 above)
Comment (hannojg) [reply package/ios/NativeWorklets.h:5]: oh haah sorry, i can put your name there, some files i was creating by hand 😂

**Comment by hannojg** [comment]
Note: i just tested these changes in the vision camera new arch PR and it compiles fine 💪

**Comment by mrousavy** [comment]
lgtm!

**Comment by mrousavy** [comment]
so can we fix C++ lint and ESLint and tests in a follow up PR?

File: .github/workflows/validate-cpp.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Run clang-format style check
-        uses: jidicula/clang-format-action@v4.13.0
+        uses: DoozyX/clang-format-lint-action@v0.18.2
```
Comment (mrousavy) [inline .github/workflows/validate-cpp.yml:32]: why?

**Comment by hannojg** [comment]
Closed in favour of:

- https://github.com/margelo/react-native-worklets-core/pull/225

**Pull request closed without merging**